### PR TITLE
[Ahuna] Add column 'scope' to AgentConfiguration

### PR DESCRIPTION
**Pull request by philipperolet**
Values 'workspace'/'published'/'private'. Related [task](https://github.com/dust-tt/dust/issues/2642)

## PR notes
Migration plan:
- deploy front-edge
- run init_db from front-edge
- deploy front

### CODE REVIEW

**Comment by philipperolet** [comment]
Actually will add the logic to update scope up to the front before merging (rather than setting workspace in lib)

File: front/lib/models/assistant/agent.ts [hunk 1]
```diff
@@ -88,6 +92,7 @@ export class AgentConfiguration extends Model<
   declare version: number;
 
   declare status: AgentStatus;
```
Comment (spolu) [inline front/lib/models/assistant/agent.ts:94]: What is Exclude ?

**Review by spolu — DISMISSED** [review]
LGTM

File: front/lib/models/assistant/agent.ts (hunk 1 above)
Comment (philipperolet) [reply front/lib/models/assistant/agent.ts:94]: New type = provided type - provided values

**Comment by philipperolet** [comment]
Updated lib & api to use scope param (it's set to 'workspace' in UI by assistant builder)

File: front/lib/assistant.ts [hunk 2]
```diff
@@ -175,8 +175,8 @@ export function compareAgentsForSort(
   if (b.sId === GLOBAL_AGENTS_SID.GPT4) return 1;
 
   // Check for agents with 'scope' set to 'workspace'
```
Comment (spolu) [inline front/lib/assistant.ts:177]: Can we update the comment above?

**Review by spolu — DISMISSED** [review]
LGTM

Modulo a minor comment update. Feel free to merge of course once fixed :+1:

File: front/lib/assistant.ts (hunk 2 above)
Comment (philipperolet) [inline front/lib/assistant.ts:177]:
```suggestion
  // Check for agents with non-global 'scope'
```

**Pull request merged**